### DISCUSSION TITLE
Water Tank Volume

### DISCUSSION THREAD

**Discussion by Johnno987** [home-assistant/frontend#16609] [Other feature requests]
I would like to see some type of gauge that is shaped like a water tank where the user can input the diameter and height of a water tank and along with a transducer connected to a Shelly Uni or Atom Lite convert a voltage into the number of litres in a water tank. This could then be represented by "litres" and a graphical image of the level of the tank? The yaml coding for this seems to be beyond my capabilities :-( 
I hope I've put this in the correct section, first time poster. Thank-you.

**Comment by spacegaier** [comment]
Gut feeling: "Water tank" is a too narrow use case to be added to the frontend and feels more like something for a custom card to cover.

What I could envision is a more generic approach of a "bar gauge" that would not be limited to a water tank but could also be used for battery levels. But even then that is not a big use case, unless I am missing some other obvious ones...

**Reply by Johnno987** [reply]
Thank-you spacegaier. And thank-you for considering my proposal. Yes, perhaps its not a big use case, but potentially I feel it could be as this method of measuring tank levels catches on (Shelly Uni and pressure transducer). I would think it would need to be specific to liquid levels though, a custom card  wouldn't even need to have a tank graphic, a regular gauge would suffice. I'm not really sure how the HA programming side of things works, but a custom card where one inputs the tanks height and diameter, a zero tank level voltage and a full level tank voltage. There are users doing this already, but the yaml is beyond me :-( sadly. 

Thanks for all you efforts with this great initiative that is HA. 

![0AFE6A6D-CDB7-4EC7-AC17-111F98683E7F](https://github.com/home-assistant/frontend/assets/116792870/315de9cf-cac9-49f0-ac0e-7ee4a984f97f)

**Reply by spacegaier** [reply]
With your last comment it doesn't sound like a graphical topic to me, as you said a regular round gauge we currently have is fine.

I think the easiest way to solve your requirement is to create a [template sensor](https://www.home-assistant.io/integrations/template/) in HA that calculates your tank level based on the inputs you mentioned (tank height, diameter, ...). You can then simply use this tank level sensor in a gauge card and you are done.

**Reply by Johnno987** [reply]
Thanks again spacegaier. I've looked at doing a template sensor, a bit beyond my capabilities at the moment. I'll do some more study and make it work. I really appreciate everything you're doing for the HA community. Have a great day :-)
